### ISSUE TITLE
Bug in number processing in tree mode

### ISSUE THREAD

**Issue by tormozit** [josdejong/jsoneditor#1553]
1. ОК
{
  "test": 123456789012
}
![изображение](https://github.com/josdejong/jsoneditor/assets/11896251/9ded14ba-2814-4e3a-956d-f56e543af981)

2. Not OK
{
  "test": 1234567890123
}
![изображение](https://github.com/josdejong/jsoneditor/assets/11896251/e7b6bfd1-10b4-44ae-97ab-b3c6aad6db68)

**Comment by josdejong** [comment]
The editor uses some heuristics to determine whether a value contains a unix timestamp, and in that case displays a tag showing the date. Similarly, it interprets when a values is a color and then shows a colorpicker.

If you don't want the date tag, you can hide it via CSS.

Do you actually have with real world data with numbers in this range?

**Comment by tormozit** [comment]
I think many users will be happy to have an option to turn off the heuristics. I agree they are usefull but not all the time. 
My case - user was confused by strange data in tree mode. And he created bug-ticket for me.

**Comment by josdejong** [comment]
Makes sense to implement a configuration option for this. And we should probably create an option for the color picker too.

Anyone interested in implementing this option? Help would be welcome.

**Comment by tormozit** [comment]
For my taste gray color is optimal for most cases. Current blue is too saturated.
`	editor.menu.style.backgroundColor = "#d0d0d0";`

Unwanted date heuristics I fixed here
```
 if (this._showTimestampTag()) {
          if (!this.dom.date) { 
          	return // FIX
            this.dom.date = document.createElement('div');
            this.dom.date.className = 'jsoneditor-date';
            this.dom.value.parentNode.appendChild(this.dom.date);
          }
```

**Comment by sridharkalaibala** [comment]
The current editor already supports disabling these tags through the documented `timestampTag` option:

```js
const editor = new JSONEditor(container, {
  mode: 'tree',
  timestampTag: false
})
editor.set({ test: 1234567890123 })
```

I checked the actual 10.4.3 browser bundle with this exact value in `tree`, `view` and `form` modes: `timestampTag: true` renders one date tag; `false` renders none; `editor.get().test` retains the original number in all six cases. No source edit or CSS workaround is needed for this behavior.

The [API documentation](https://github.com/josdejong/jsoneditor/blob/85dfeee8129f2a47ca67d8475327f9e3716797d5/docs/api.md#configuration-options) also describes a callback for deciding per field/value/path. If `timestampTag: false` still shows a tag in your integration, please share the JSONEditor version and constructor options so we can investigate that specific case.
